ci/common-ci.yml

```diff
@@ -34,7 +34,7 @@ stages:
     reports:
       dotenv: build.env
   variables:
-    SPACK_SHA: develop-2024-10-06
+    SPACK_SHA: develop-2024-11-10
```

Comment:
Interestingly spack concretizes the environment with pika 0.29.0 even though 0.30.0 is available in 2024-11-10... We do need a change in the spack package anyway to require 0.30:. I'll push a change for that.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/1125/commits/43763dc2fe86ff12a730ad7d86186128e79e70ae